api/src/main/java/com/community/api/services/QualificationDetailsService.java

```diff
@@ -11,166 +14,218 @@
-       CustomCustomer customCustomer= entityManager.find(CustomCustomer.class,customCustomerId);
-       if(customCustomer==null)
-       {
-           throw new CustomerDoesNotExistsException("Customer does not exist with id "+ customCustomerId);
-       }
-       List<QualificationDetails> qualificationDetails = customCustomer.getQualificationDetailsList();
-       QualificationDetails qualificationDetailsToUpdate =null;
-       for(QualificationDetails qualificationDetails1 : qualificationDetails)
-       {
-           if(qualificationDetails1.getId()==qualificationId)
-           {
-               qualificationDetailsToUpdate = qualificationDetails1;
-               break;
-           }
-       }
-       if (qualificationDetailsToUpdate == null) {
-           throw new EntityDoesNotExistsException("Qualification details with id " + qualificationId+ " does not exists");
-       }
-
-       TypedQuery<QualificationDetails> query = entityManager.createQuery(
-               "SELECT q FROM QualificationDetails q WHERE q.custom_customer.id = :customerId AND q.qualification_id = :qualification_id",
-               QualificationDetails.class
-       );
-       query.setParameter("customerId", customCustomerId);
-       query.setParameter("qualification_id", qualification.getQualification_id());
-       QualificationDetails existingQualificationDetails = query.getResultStream().findFirst().orElse(null);
-
-       if (existingQualificationDetails != null && qualificationId!= existingQualificationDetails.getId()) {
-           throw new EntityAlreadyExistsException("Qualification details with id " + qualification.getQualification_id() + " already exists");
-       }
-
-       if (Objects.nonNull(qualification.getQualification_id())) {
-           List<DocumentType> qualifications = qualificationService.getAllQualifications();
-           Integer examinationToAdd = null;
-
-           for (DocumentType examination : qualifications) {
-               if ((examination.getDocument_type_id())==qualification.getQualification_id()) {
-                   examinationToAdd = examination.getDocument_type_id();
-                   break;
-               }
-           }
-
-           if (examinationToAdd == null) {
-               throw new ExaminationDoesNotExistsException("Qualification with id " + qualification.getQualification_id() + " does not exist");
-           }
-           qualificationDetailsToUpdate.setQualification_id(examinationToAdd);
-       }
+    @Transactional
+    public QualificationDetails updateQualificationDetail(Long userId, Long qualificationId, UpdateQualificationDto qualification, String roleName) throws EntityDoesNotExistsException, EntityAlreadyExistsException, CustomerDoesNotExistsException, ExaminationDoesNotExistsException {
+        List<QualificationDetails> qualificationDetails;
```

Comment:
here if someone enters string
give NumberFormatException exception

use this
Long id = Long.valueOf(customerId);

catch (NumberFormatException e) {
    return ResponseService.generateErrorResponse("Invalid customerId: expected a Long", HttpStatus.BAD_REQUEST);
}